components/BlurhashCanvas.vue

```diff
@@ -36,16 +36,16 @@ export default Vue.extend({
     this.draw();
   },
   methods: {
-    draw() {
-      const pixels = decode(this.hash, this.width, this.height);
-      if (pixels) {
-        const ctx = (this.$refs.canvas as HTMLCanvasElement).getContext('2d');
-        const imageData = ctx?.createImageData(this.width, this.height);
+    async draw() {
+      const ctx = (this.$refs.canvas as HTMLCanvasElement).getContext('2d');
+      const imageData = ctx?.createImageData(this.width, this.height);
+      try {
+        const pixels = await getPixels(this.hash, this.width, this.height);
         if (imageData) {
           imageData.data.set(pixels);
           ctx?.putImageData(imageData, 0, 0);
         }
-      }
+      } catch {}
```

Comment:
#306 Hides the canvas if the image is not valid

Reply:
@Camc134 What about setting validHash in the catch?

I want to keep the try-catch block so whatever error that could happen in the worker doesn't hang the component lifecycle and it's printed in the console. Detecting if the Blurhash is valid with isBlurhashValid is also more expensive than doing it this way (since a "partial" decode happens) and not always returns false when it should (I have one item that has a corrupt Blurhash due to a corrupted image, so I tested all of this).

Try-catch is as inexpensive as it could get in the main thread.

Reply:
That's fine. I was just leaving the comment there as a note. When I update that PR, I'll update uit